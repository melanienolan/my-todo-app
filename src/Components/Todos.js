import React, { Component } from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import TodoItem from './TodoItem'; // eslint-disable-line no-unused-vars

const Todos = props => {
  let todoItems;
  if (props.todos) {
    todoItems = props.todos.map(todo => {
      return (
        <TodoItem
          todo={todo}
          key={todo.id}
          onCompleted={() => props.onCompleted(todo)}
          onDeleted={() => props.onDeleted(todo)}
        />
      );
    });
  }
  return (
    <div>
      {todoItems}
    </div>
  );
};

Todos.propTypes = {
  todos: PropTypes.array,
  onCompleted: PropTypes.func,
  onDeleted: PropTypes.func
};

export default Todos;
