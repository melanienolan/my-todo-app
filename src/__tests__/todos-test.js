import React from 'react';
import Todos from '../Components/Todos';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const todos = [
  {
    completed: true,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'complete todo'
  },
  {
    completed: false,
    id: 'e41fac97-3d2d-457a-9990-479f016ad0ce',
    title: 'incomplete todo'
  }
];

it('renders todos component without crashing', () => {
  shallow(<Todos todos={todos} />);
});

it('renders todo component with todo items', () => {
  const todosComponent = shallow(<Todos todos={todos} />);
  expect(toJson(todosComponent)).toMatchSnapshot();
});
