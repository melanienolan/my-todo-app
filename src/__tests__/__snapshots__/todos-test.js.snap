// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders todo component with todo items 1`] = `
<CSSTransitionGroup
  className="todo-list"
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={200}
  transitionLeave={true}
  transitionLeaveTimeout={200}
  transitionName="grow"
>
  <TodoItem
    onCompleted={[Function]}
    onDeleted={[Function]}
    todo={
      Object {
        "completed": true,
        "id": "e41fac97-3d2d-457a-9990-479f016ad0ce",
        "title": "complete todo",
      }
    }
  />
  <TodoItem
    onCompleted={[Function]}
    onDeleted={[Function]}
    todo={
      Object {
        "completed": false,
        "id": "e41fac97-3d2d-457a-9990-479f016ad0ce",
        "title": "incomplete todo",
      }
    }
  />
</CSSTransitionGroup>
`;
